<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.8.13">
  <compounddef id="structxn_1_1XNetworkAlgorithmError" kind="struct" language="C++" prot="public">
    <compoundname>xn::XNetworkAlgorithmError</compoundname>
    <basecompoundref refid="structxn_1_1XNetworkException" prot="public" virt="non-virtual">xn::XNetworkException</basecompoundref>
    <derivedcompoundref refid="structxn_1_1XNetworkUnbounded" prot="public" virt="non-virtual">xn::XNetworkUnbounded</derivedcompoundref>
    <derivedcompoundref refid="structxn_1_1XNetworkUnfeasible" prot="public" virt="non-virtual">xn::XNetworkUnfeasible</derivedcompoundref>
    <includes refid="exception_8hpp" local="no">exception.hpp</includes>
      <sectiondef kind="public-func">
      <memberdef kind="function" id="structxn_1_1XNetworkAlgorithmError_1afe29660b0b98f56d1d36027c0dfbb16c" prot="public" static="no" const="no" explicit="yes" inline="yes" virt="non-virtual">
        <type></type>
        <definition>xn::XNetworkAlgorithmError::XNetworkAlgorithmError</definition>
        <argsstring>(const char *msg)</argsstring>
        <name>XNetworkAlgorithmError</name>
        <param>
          <type>const char *</type>
          <declname>msg</declname>
        </param>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/media/lubuntu/USBDISK/github/ckpttncpp/lib/include/xnetwork/exception.hpp" line="79" column="1" bodyfile="/media/lubuntu/USBDISK/github/ckpttncpp/lib/include/xnetwork/exception.hpp" bodystart="79" bodyend="82"/>
      </memberdef>
      </sectiondef>
    <briefdescription>
    </briefdescription>
    <detaileddescription>
<para>Exception for unexpected termination of algorithms. </para>    </detaileddescription>
    <inheritancegraph>
      <node id="287">
        <label>xn::XNetworkUnfeasible</label>
        <link refid="structxn_1_1XNetworkUnfeasible"/>
        <childnode refid="283" relation="public-inheritance">
        </childnode>
      </node>
      <node id="284">
        <label>xn::XNetworkException</label>
        <link refid="structxn_1_1XNetworkException"/>
        <childnode refid="285" relation="public-inheritance">
        </childnode>
      </node>
      <node id="283">
        <label>xn::XNetworkAlgorithmError</label>
        <link refid="structxn_1_1XNetworkAlgorithmError"/>
        <childnode refid="284" relation="public-inheritance">
        </childnode>
      </node>
      <node id="285">
        <label>std::runtime_error</label>
      </node>
      <node id="289">
        <label>xn::XNetworkNoPath</label>
        <link refid="structxn_1_1XNetworkNoPath"/>
        <childnode refid="287" relation="public-inheritance">
        </childnode>
      </node>
      <node id="286">
        <label>xn::XNetworkUnbounded</label>
        <link refid="structxn_1_1XNetworkUnbounded"/>
        <childnode refid="283" relation="public-inheritance">
        </childnode>
      </node>
      <node id="288">
        <label>xn::XNetworkNoCycle</label>
        <link refid="structxn_1_1XNetworkNoCycle"/>
        <childnode refid="287" relation="public-inheritance">
        </childnode>
      </node>
    </inheritancegraph>
    <collaborationgraph>
      <node id="291">
        <label>xn::XNetworkException</label>
        <link refid="structxn_1_1XNetworkException"/>
        <childnode refid="292" relation="public-inheritance">
        </childnode>
      </node>
      <node id="290">
        <label>xn::XNetworkAlgorithmError</label>
        <link refid="structxn_1_1XNetworkAlgorithmError"/>
        <childnode refid="291" relation="public-inheritance">
        </childnode>
      </node>
      <node id="292">
        <label>std::runtime_error</label>
      </node>
    </collaborationgraph>
    <location file="/media/lubuntu/USBDISK/github/ckpttncpp/lib/include/xnetwork/exception.hpp" line="78" column="1" bodyfile="/media/lubuntu/USBDISK/github/ckpttncpp/lib/include/xnetwork/exception.hpp" bodystart="77" bodyend="83"/>
    <listofallmembers>
      <member refid="structxn_1_1XNetworkAlgorithmError_1afe29660b0b98f56d1d36027c0dfbb16c" prot="public" virt="non-virtual"><scope>xn::XNetworkAlgorithmError</scope><name>XNetworkAlgorithmError</name></member>
      <member refid="structxn_1_1XNetworkException_1adf7ac46d6cef6d2fcee880fc2f709458" prot="public" virt="non-virtual"><scope>xn::XNetworkAlgorithmError</scope><name>XNetworkException</name></member>
    </listofallmembers>
  </compounddef>
</doxygen>
